### PR TITLE
Update Debian packaging for Recursor

### PR DESCRIPTION
**Pull request by zeha**
### Short description
Resync packaging with Debian.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
I have zero knowledge on how debian packages works, so it is not possible for me to review this PR.

**Comment by Habbie** [comment]
> I have zero knowledge on how debian packages works, so it is not possible for me to review this PR.

I requested a review from me.

File: builder-support/debian/recursor/debian-buster/control [hunk 1]
```diff
@@ -13,8 +13,8 @@ Build-Depends: debhelper-compat (= 10),
                libboost-thread-dev,
                libcap-dev [linux-any],
                libfstrm-dev,
-               libluajit-5.1-dev [amd64] | liblua5.1-0-dev,
-               liblua5.1-0-dev [!amd64],
+               libluajit-5.1-dev [amd64] | liblua5.3-dev,
```
Comment (Habbie) [inline builder-support/debian/recursor/debian-buster/control:16]: I suppose we want to replace this with the version in auth.

File: builder-support/debian/recursor/debian-buster/control [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
+               libcap-dev [linux-any],
                libfstrm-dev,
+               libluajit-5.1-dev [amd64] | liblua5.3-dev,
+               liblua5.3-dev [!amd64],
+               libprotobuf-dev,
+               libsnmp-dev,
                libsodium-dev,
                libssl-dev,
-               libsystemd-dev [linux-any],
+               libsystemd-dev,
```
Comment (Habbie) [inline builder-support/debian/recursor/debian-buster/control:22]: Is this change missing in the auth?

File: builder-support/debian/recursor/debian-buster/rules [hunk 3]
```diff
@@ -1,76 +1,63 @@
...
+export DEB_BUILD_MAINT_OPTIONS = hardening=+all
+# see EXAMPLES in dpkg-buildflags(1) and read /usr/share/dpkg/*
 DPKG_EXPORT_BUILDFLAGS = 1
-# Include buildflags.mk so we can append to the vars it sets.
-include /usr/share/dpkg/buildflags.mk
+include /usr/share/dpkg/default.mk
 
-# Only enable systemd integration on Linux operating systems
+# Build with libcap on linux.
 ifeq ($(DEB_HOST_ARCH_OS),linux)
```
Comment (Habbie) [inline builder-support/debian/recursor/debian-buster/rules:11]: Why check for Linux here if the check is gone for systemd?

**Review by Habbie — CHANGES_REQUESTED** [review]
Couple of questions, LGTM otherwise

File: builder-support/debian/recursor/debian-buster/control (hunk 2 above)
Comment (zeha) [reply builder-support/debian/recursor/debian-buster/control:22]: Indeed. Also missing in Debian.

File: builder-support/debian/recursor/debian-buster/control (hunk 1 above)
Comment (zeha) [reply builder-support/debian/recursor/debian-buster/control]: Indeed, done.

File: builder-support/debian/recursor/debian-buster/rules (hunk 3 above)
Comment (zeha) [reply builder-support/debian/recursor/debian-buster/rules:11]: Removed, thanks. Also applied in Debian

**Comment by zeha** [comment]
debhelper >= 10 requires stretch and newer, notably won't work on xenial without xenial-backports. Should be fine for master.

**Comment by omoerbeek** [comment]
It is unclear what the status is of this PR, even after the conflicts would be resolved:

> This undoes #8639. 

Is that a problem?

More general is this PR still relevant? If so, please rebase.

**Comment by omoerbeek** [comment]
Any news on this? @zeha @Habbie ?

**Comment by zeha** [comment]
> > This undoes https://github.com/PowerDNS/pdns/pull/8639.
> Is that a problem?

I think it's the correct behaviour, but IIRC 8639 was done because of a support request on github.

**Comment by Habbie** [comment]
We accept that this reverts 8639. Can you update the PR?

**Comment by omoerbeek** [comment]
Rebased

**Comment by zeha** [comment]
I rechecked the diff against `debian/4.8.2-1`, and it seems good to me.

**Pull request merged**